R/utils.R

```diff
@@ -770,22 +770,47 @@ formatNoSci <- function(x) {
   format(x, scientific = FALSE, digits = 15)
 }
 
+
+# This function when called without arguments returns the mtime of the most
+# recently changed file watched by the autoreload process. When given `value`, a
+# (possibly) vector of file modification times, it stores the max time seen so
+# far. This value is used by `cachedFuncWithFile()` when autoreload is enabled
+# to reload app/ui/server files when watched supporting files are changed.
+cachedAutoReloadMostRecentChange <- local({
+  last_update <- 0
+
+  max_updated <- function(value) {
+    max(suppressWarnings(max(value, last_update, na.rm = TRUE)), 0)
+  }
+
+  function(value = NULL) {
+    if (!is.null(value) && !is.na(value) && length(value)) {
+      last_update <<- max_updated(value)
+      return(invisible(value))
```

Comment:
Should this be `return(invisible(last_update))`?

Reply:
I rewrote this as getter/setter closures (in da9ca3c5b1054546e26abd1ab0a977172960d6e8), which avoids finding the max changed time and instead tracks the file change update.